File: src/trading_ai/ui/static/executive_reporting.js
(() => {
  const api=async(url,opt={})=>{
    const r=await fetch(url,{headers:{"Content-Type":"application/json"},cache:"no-store",...opt});
    const b=await r.json().catch(()=>({}));
    if(!r.ok)throw new Error(b.detail||JSON.stringify(b));
    return b;
  };
  const n=(v,d=2)=>Number(v||0).toFixed(d);
  function ensureNav(){
    const root=document.querySelector("#nav")||document.querySelector(".nav");
    if(root&&!root.querySelector('[data-view="executive-reporting"]')){
      const b=document.createElement("button");
      b.dataset.view="executive-reporting";
      b.textContent="Executive Dashboard";
      root.appendChild(b);
    }
  }
  async function render(){
    const content=document.querySelector("#content");
    if(document.querySelector("#pageTitle"))document.querySelector("#pageTitle").textContent="Executive Dashboard & Institutional Reporting";
    if(document.querySelector("#notice"))document.querySelector("#notice").textContent="Read-only KPI scorecards, board analytics, institutional summaries, and export-ready compliance evidence.";
    content.innerHTML="<p>Loading executive reporting artifacts…</p>";
    try{
      const [score,board]=await Promise.all([
        api("/api/v1/executive/scorecard"),
        api("/api/v1/executive/board-report")
      ]);
      content.innerHTML=`
      <div class="option-chain-summary">
        <div class="option-chain-card"><small>Net P/L</small><strong>${n(score.total_net_pnl)}</strong></div>
        <div class="option-chain-card"><small>Win Rate</small><strong>${n(score.win_rate*100)}%</strong></div>
        <div class="option-chain-card"><small>Sharpe</small><strong>${n(score.sharpe_ratio)}</strong></div>
        <div class="option-chain-card"><small>Max Drawdown</small><strong>${n(score.max_drawdown*100)}%</strong></div>
        <div class="option-chain-card"><small>Active Incidents</small><strong>${score.active_incidents}</strong></div>
        <div class="option-chain-card"><small>Critical Alerts</small><strong>${score.critical_alerts}</strong></div>
      </div>
      <div class="card"><h2>KPI Scorecard</h2>${kpis(score.kpis)}</div>
      <div class="card"><h2>Board-Level Analytics</h2><p>${board.executive_summary}</p>${sections(board.sections)}</div>
      <div class="card"><h2>Regulatory Export</h2>
        <label>Export Type<select id="execExportType">
          <option>RISK_SUMMARY</option><option>EXECUTION_ACTIVITY</option><option>GOVERNANCE_AUDIT</option><option>ACCESS_REVIEW</option><option>FULL_EVIDENCE_PACKAGE</option>
        </select></label>
        <button id="execExport">Generate Export</button><pre id="execExportResult"></pre>
      </div>
      ${score.warnings.length?`<div class="warning">${score.warnings.join("<br>")}</div>`:""}`;
      document.querySelector("#execExport").onclick=async()=>{
        try{
          const result=await api("/api/v1/executive/regulatory-exports",{method:"POST",body:JSON.stringify({
            export_type:document.querySelector("#execExportType").value,
            start_date:null,end_date:null,include_source_paths:true
          })});
          document.querySelector("#execExportResult").textContent=JSON.stringify(result,null,2);
        }catch(e){document.querySelector("#execExportResult").textContent=e.message}
      };
    }catch(e){content.innerHTML=`<div class="error">${e.message}</div>`}
  }
  const kpis=rows=>`<div style="overflow:auto"><table><thead><tr><th>KPI</th><th>Value</th><th>Unit</th><th>Status</th><th>Source</th></tr></thead><tbody>${rows.map(x=>`<tr><td>${x.label}</td><td>${n(x.value,4)}</td><td>${x.unit}</td><td>${x.status}</td><td>${x.source}</td></tr>`).join("")}</tbody></table></div>`;
  const sections=rows=>rows.map(x=>`<div class="option-chain-card"><h3>${x.title}</h3><p>${x.summary}</p>${x.highlights.length?`<strong>Highlights</strong><ul>${x.highlights.map(v=>`<li>${v}</li>`).join("")}</ul>`:""}${x.risks.length?`<strong>Risks</strong><ul>${x.risks.map(v=>`<li>${v}</li>`).join("")}</ul>`:""}</div>`).join("");
  window.addEventListener("DOMContentLoaded",()=>{ensureNav();if(new URLSearchParams(location.search).get("view")==="executive-reporting")render()});
  document.addEventListener("click",e=>{if(e.target.closest('[data-view="executive-reporting"]')){history.pushState({},"","/?view=executive-reporting");render()}});
})();

File: src/trading_ai/ui/models/executive_reporting.py
from __future__ import annotations

from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, Field


class KpiMetric(BaseModel):
    metric_id: str
    label: str
    value: float
    unit: str = ""
    status: Literal["GOOD", "WATCH", "BAD", "UNKNOWN"] = "UNKNOWN"
    period: str = "CURRENT"
    source: str = ""


class ExecutiveScorecard(BaseModel):
    generated_at: datetime
    as_of_date: date
    total_net_pnl: float = 0
    gross_exposure: float = 0
    net_exposure: float = 0
    win_rate: float = 0
    sharpe_ratio: float = 0
    max_drawdown: float = 0
    active_strategies: int = 0
    active_incidents: int = 0
    critical_alerts: int = 0
    open_access_reviews: int = 0
    kpis: list[KpiMetric] = Field(default_factory=list)
    warnings: list[str] = Field(default_factory=list)


class BoardSection(BaseModel):
    title: str
    summary: str
    highlights: list[str] = Field(default_factory=list)
    risks: list[str] = Field(default_factory=list)


class BoardReport(BaseModel):
    generated_at: datetime
    title: str
    reporting_period: str
    executive_summary: str
    sections: list[BoardSection]
    approvals: list[str] = Field(default_factory=list)
    source_artifacts: list[str] = Field(default_factory=list)


class RegulatoryExportRequest(BaseModel):
    export_type: Literal[
        "RISK_SUMMARY",
        "EXECUTION_ACTIVITY",
        "GOVERNANCE_AUDIT",
        "ACCESS_REVIEW",
        "FULL_EVIDENCE_PACKAGE",
    ]
    start_date: date | None = None
    end_date: date | None = None
    include_source_paths: bool = True


class RegulatoryExportRecord(BaseModel):
    export_id: str
    export_type: str
    generated_at: datetime
    start_date: date | None = None
    end_date: date | None = None
    record_count: int = 0
    checksum: str
    output_path: str
    warnings: list[str] = Field(default_factory=list)
